### PR TITLE
feat: convert preferences to support library

### PR DESCRIPTION
**Pull request by david-allison-1**
This needs to be done in a single (large) commit

```
convert preference_headers.xml to <PreferenceScreen>
(XML) Use support library compatible classes for our classes:
  ConfirmationPreference -> ConfirmationPreferenceCompat
  NumberRangePreference -> NumberRangePreferenceCompat
  IncrementerNumberRangePreference -> IncrementerNumberRangePreferenceCompat
  CustomDialogPreference -> ResetLanguageDialogPreference
    also added "key"
  SeekBarPreference -> SeekBarPreferenceCompat
(Preferences.java)
  Use Fragments
    Convert from AppCompatPreferenceActivity to AnkiActivity
    Define a layout for the activity
  android.preference -> androidx.preference
    move back to imports of androidx.preference
    fully qualified names (android.preference) were used as
    deprecated imports caused unavoidable compiler warnings
  Handle intents to allow intent-based calls to still work
    Example: Open advanced prefs if the collection can't be accessed
  define onDisplayPreferenceDialog to allow our custom preferences to work
    https://stackoverflow.com/questions/5533078/timepicker-in-preferencescreen/34398747#34398747
```

Remove a few methods and classes which are no longer used

#5019 for Preferences.java, not for DeckOptions

## How Has This Been Tested?

On my Android 11 device

⚠️ May need an API 21 test

## Checklist
- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/Preferences.java [hunk 1]
```diff
@@ -226,7 +242,7 @@ public void onBackPressed() {
         }
     }
 
-
+    @SuppressWarnings("deprecation") // startActivity
```
Comment (mikehardy) [inline AnkiDroid/src/main/java/com/ichi2/anki/Preferences.java:245]: one at a time google, one at a time...

File: AnkiDroid/src/main/java/com/ichi2/anki/Preferences.java [hunk 2]
```diff
@@ -568,17 +582,42 @@ public void onSharedPreferenceChanged(SharedPreferences sharedPreferences, Strin
             updatePreference(((Preferences) getActivity()), sharedPreferences, key);
         }
 
+
+        @Override
+        @SuppressWarnings("deprecation") // setTargetFragment
```
Comment (mikehardy) [inline AnkiDroid/src/main/java/com/ichi2/anki/Preferences.java:587]: Migrate away from setTargetFragment

We aren't tracking this deprecation yet, I'll peel this comment to a new issue

This contains a sketch of an implementation (note a link to a comment on an answer - the comment refines the answer in a necessary way I think):
https://stackoverflow.com/questions/64869501/how-to-replace-settargetfragment-now-that-it-is-deprecated#comment114721602_64869730 

Here's the deprecated API reference:
https://developer.android.com/reference/androidx/fragment/app/Fragment#setTargetFragment(androidx.fragment.app.Fragment,%20int)

Comment (david-allison-1) [reply AnkiDroid/src/main/java/com/ichi2/anki/Preferences.java:587]: Ah, I've just done this in the csv code. Wasn't difficult at all

File: AnkiDroid/src/main/res/layout/preferences.xml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+    <include layout="@layout/toolbar" />
+
+    <FrameLayout
+        android:id="@+id/settings_container"
+        android:layout_width="0dp"
+        android:layout_height="0dp"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintTop_toBottomOf="@id/toolbar"
```
Comment (mikehardy) [inline AnkiDroid/src/main/res/layout/preferences.xml:32]: Only joking but I imagine we're failing to handle BTT (Bottom-to-Top) languages correctly here, will probably need to revise in future

**Review by mikehardy — APPROVED** [review]
I'll pull locally and test on the emulator gamut

**Comment by mikehardy** [comment]
Everything appeared to work great on API 21, 28, 30 (picked sorta random but low and high basically). Focused especially on 21 setting all sorts of prefs (especially the ones that make you open up sub-dialogs / sub-screens etc) then making sure they where there. :trophy:

**Comment by david-allison-1** [comment]
So happy to see the end of this one, thanks!

**Pull request merged**